skbio/diversity/alpha/tests/test_base.py

```diff
@@ -393,6 +394,20 @@ def test_observed_otus(self):
     def test_osd(self):
         self.assertEqual(osd(self.counts), (9, 3, 3))
 
+    def test_pielou(self):
+        # Calculate "by hand".
+        arr = np.array([1, 2, 3, 1])
+        h = shannon(arr)
+        s = 4
+        expected = h / np.log(s)
+        self.assertAlmostEqual(pielou(arr), expected)
+
+        self.assertAlmostEqual(pielou(self.counts), 1.334283585856)
```

Comment:
@ebolyen @gregcaporaso what do you think about having `nan` returned for the divide-by-zero case? I don't know whether it's more useful to return that or have it raise an error saying the metric is undefined when there are no observed OTUs.

Regarding empty input `[]`, all metrics should currently be raising that error since it happens within `_validate_counts_vector`. I think `[]` should be explicitly disallowed because there are essentially no data for a metric to operate on. A better error message should be raised though. @ebolyen @gregcaporaso @colinbrislawn do you agree? I don't think this should get addressed in this PR since it has a broader scope.